pkg/apiserver/resthandler.go

```diff
@@ -703,6 +703,99 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 	}
 }
 
+// DeleteCollection returns a function that will handle a collection deletion
+func DeleteCollection(r rest.CollectionDeleter, checkBody bool, scope RequestScope, admit admission.Interface) restful.RouteFunction {
+	return func(req *restful.Request, res *restful.Response) {
+		w := res.ResponseWriter
+
+		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
+		timeout := parseTimeout(req.Request.URL.Query().Get("timeout"))
+
+		namespace, err := scope.Namer.Namespace(req)
+		if err != nil {
+			errorJSON(err, scope.Codec, w)
+			return
+		}
+
+		ctx := scope.ContextFunc(req)
+		ctx = api.WithNamespace(ctx, namespace)
+
+		if admit != nil && admit.Handles(admission.Delete) {
+			userInfo, _ := api.UserFrom(ctx)
+
+			err = admit.Admit(admission.NewAttributesRecord(nil, scope.Kind.Kind, namespace, "", scope.Resource.Resource, scope.Subresource, admission.Delete, userInfo))
```

Comment:
I am wondering if admission control should make it clearer when you are doing a delete over an entire collection rather than just DELETE + kind + (optional namespace).  It's probably fine as is, but will mull on it some.